src/java/org/apache/cassandra/config/Config.java

```diff
@@ -1149,6 +1151,13 @@ public enum DiskAccessMode
         standard,
     }
 
+    public enum CommitLogDiskAccessMode
```

Comment:
Maybe we can avoid creating another enum? The existing `DiskAccessMode` can be probably used as well with added `direct` option and some validation in places where it is used whether the selected access mode is supported or not? (TBH, I'd rather remove `mmap_index_only` from `DiskAccessMode` to make it more generic - disk access mode for indexes could be specified explicitly instead)

Reply:
Is this flexible? As directio is only work for commitlog write, It May not be supported  for normal sstable read /write and commitlog read and others situation yet if DiskAccessMode is direct_io.

Reply:
I meant we could simply validate the configuration either in `DatabaseDescriptor` or where the files are opened.

Reply:
Good idea. Shall I add direct_io to existing enum and change the relavant functions to verify the correct input?